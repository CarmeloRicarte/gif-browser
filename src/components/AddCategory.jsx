import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  /**
   * When the input changes, set the input value to the value of the input.
   */
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  /**
   * The function onSubmit calls the function onNewCategory, which is passed in as a prop from the
   * parent component.
   * The function onNewCategory is passed the value of the input field.
   * @returns The input value is being returned.
   */
  const onSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) {
      return;
    }
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        id="seachGifs"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
