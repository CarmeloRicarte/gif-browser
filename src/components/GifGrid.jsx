import { useEffect, useState } from "react";
import { getGifs } from "./../helpers/getGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  /**
   * GetImages is an async function that calls getGifs, which is an async function that returns a
   * promise, and then sets the state of images to the value of the promise
   */
  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h3>{category}</h3>

      <div>
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
