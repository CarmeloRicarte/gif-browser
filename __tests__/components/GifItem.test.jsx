import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Tests of <GifItem />", () => {
  test("should match with snapshot", () => {
    const title = "Dragon Ball Ultra Instinct GIF by Toei Animation";
    const url =
      "https://media3.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=947a36cfqz7o481dq7dqg783yfc0888b1zif61sx5u761lrp&rid=giphy.gif&ct=g";

    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
