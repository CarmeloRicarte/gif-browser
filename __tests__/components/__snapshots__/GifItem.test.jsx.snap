// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Ultra Instinct GIF by Toei Animation"
      src="https://media3.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=947a36cfqz7o481dq7dqg783yfc0888b1zif61sx5u761lrp&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball Ultra Instinct GIF by Toei Animation
    </p>
  </div>
</div>
`;
